### ISSUE TITLE
New Exception: Linux Kernel Enforcement Statement

### ISSUE THREAD

**Issue by karensandler** [spdx/license-list-XML#655]
The Linux Kernel Enforcement Statement, an additional permission on GPL-2.0-only, has been gaining increased adoption. In April, I agreed in response to a discussion at a legal and licensing related meeting (where many SPDX contributors were present) to submit this for acceptance (sorry for the delay!).

Full Name: Linux Kernel Enforcement Statement for GPL-2.0-only

Short Identifier: Linux-Termination-Addl-Permission

There's version of the [exception text](https://www.kernel.org/doc/html/latest/process/kernel-enforcement-statement.html), which is formatted from the version that [appears in Linus' Git tree for Linux](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/process/kernel-enforcement-statement.rst) (It's also available [here on GitHub](https://github.com/torvalds/linux/blob/master/Documentation/process/kernel-enforcement-statement.rst).)

**Comment by jlovejoy** [comment]
@karensandler - I believe we discussed this on the SPDX legal team email list,  as well as the Red Hat (and other company) GPLv3 Cure Commitments and determined it was not needed/appropriate to add here due to being copyright-holder specific (going off of memory, will have to look back at notes!). 
(I'm (obviously) all for spreading the word on this otherwise!)
let me look back and might be good to have a revisit on the next legal call.

**Comment by bkuhn** [comment]
@jlovejoy wrote:
> - I believe we discussed this on the SPDX legal team email list, as well as the Red Hat (and other company) GPLv3 Cure Commitments and determined it was not needed/appropriate to add here due to being copyright-holder specific (going off of memory, will have to look back at notes!).

I can't give you a direct link to the SPDX mailing list archives on this, as all historical archive links broke completely recently, but I distinctly recall that there were separate issues about the two exceptions, and the discussion was cut short, at your request, due the pending 3.0 release.  I recall that you said we should table it to after 3.0.

Anyway, the Linux Kernel Enforcement Statement is certainly project-specific (the name "Linux" is right there in the name, obviously), but not *copyright holder* specific, since dozens of copyright holders have signed on.

I don't have a strong opinion on Red Hat's statement, but it is drafted somewhat differently and therefore presumably should be analyzed differently so as not to conflate apples and oranges.

**Comment by bkuhn** [comment]
This was discussed on the SPDX Legal call today.  @jlovejoy  summed up at the end of the call saying: "everyone on the call was supportive of adding them [this one and the RH one]".  @jlovejoy  indicated that this ticket should remain open until the next release goes out next week with this additional permission included.

It was agreed at the end of the call that this will be added in the next week.  An opportunity will be provided to object on the list by providing an explanatory legal theory that explains that this additional permission is somehow fundamentally different from other additional permissions already on the list.

**Comment by jlovejoy** [comment]
re: naming and short identifier: would it make more sense to use the following for the short identifier to be more consistent with full name and name in actual use?

Full Name: Linux Kernel Enforcement Statement for GPL-2.0-only

Short Identifier: Linux-enforcement-statement

**Comment by bkuhn** [comment]
> Full Name: Linux Kernel Enforcement Statement for GPL-2.0-only

Note that there is no reason someone couldn't use this form of additional permission for a GPL-2.0-or-later project.  Someone indeed might want GPL-2.0-or-later WITH Linux-enforcement-statement , because downstream would be permitted to narrow to GPL-2.0-only WITH Linux-enforcement-statement.
The question might be "well, why would someone do that, when they could do it with the RH drafting".  I'll be frank that I prefer the Linux drafting to the RH drafting of a similar additional permission.  The differences are minor but salient.

> Short Identifier: Linux-enforcement-statement

It seems a bit long for a short id?

**Comment by richardfontana** [comment]
I agree with @bkuhn that the full name of the identifier should not be limited to GPL-2.0-only. I don't think  the GPL needs to be referenced in the name (I don't think this is done with any existing SPDX exceptions).

**Comment by jlovejoy** [comment]
All of the commentary on this occurred on the mailing list (which is fine) but this recent activity made me realize that we never put a summary here, which we've been trying to do: 
TL;DR - Given that the people who are actually working in implementing SPDX identifiers in the kernel were not asking for or wanting it and there seemed to be a lot of controversy and nothing had changed since it was first considered to warrant a change - no change to the current status (i.e. no identifier for this) was the outcome then. 
see https://lists.spdx.org/g/Spdx-legal/message/2499 for full summary.

This issue should have been closed then, but I kept it as "later release" due to feeling pressured to do so.  I'm closing it now, as I don't see that anything has changed to reverse the above decision.

**Comment by pombredanne** [comment]
I am looking at all the licenses detected in the kernel and not listed in the main SPDX license list to submit them as discussed and I am not clear why this would be closed: this is detected by ScanCode alright when you scan the kernel and has been for a long while. And the same applies to #714 (e.g. the GPL-CC).
This sounds a bit weird to me. Let me post on list to get some feedback there instead.

**Comment by bonzini** [comment]
Maybe the issue here is that the proposal concerned the whole KES, and not just the additional permissions that it adds on top of the GPLv2.0? As someone who signed the KES but doesn't own any copyright in Linux I can see that it is a somewhat strange beast, at the same time if I were to start a GPLv2+ project now I would probably consider including the KES additional permissions in the license.

This would be this text:

    Notwithstanding the termination provisions of the GPL-2.0, we agree that
    it is in the best interests of our development community to adopt the
    following provisions of GPL-3.0 as additional permissions under our
    license with respect to any non-defensive assertion of rights under the
    license.

        However, if you cease all violation of this License, then your license
        from a particular copyright holder is reinstated (a) provisionally,
        unless and until the copyright holder explicitly and finally
        terminates your license, and (b) permanently, if the copyright holder
        fails to notify you of the violation by some reasonable means prior to
        60 days after the cessation.

        Moreover, your license from a particular copyright holder is
        reinstated permanently if the copyright holder notifies you of the
        violation by some reasonable means, this is the first time you have
        received notice of violation of this License (for any work) from that
        copyright holder, and you cure the violation prior to 30 days after
        your receipt of the notice.

**Comment by pombredanne** [comment]
@bonzini you wrote:
> This would be this text:
```
Notwithstanding the termination provisions of the GPL-2.0, we agree that
[...]
your receipt of the notice.
```
... exactly. I do not care too much for the commentary above the (evolving) list of committers below which are more context than core... but the statement you highlighted deserves to be detected and IMHO deserves a name of its own.

**Comment by jlovejoy** [comment]
in the case of a new project wanting to apply GPL-3.0 termination clauses, there is Red Hat's GPL Cooperation Commitment for projects, which is designed and drafted for this purpose.

**Comment by richardfontana** [comment]
@bonzini see: https://gplcc.github.io/gplcc/Project/README-PROJECT.html
